pkg/geoipupdate/download/download.go

```diff
@@ -0,0 +1,149 @@
+// Package download provides a library for checking/downloading/updating mmdb files.
+package download
+
+import (
+	"context"
+	"crypto/md5"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+const (
+	Extension = ".mmdb"
+
+	// zeroMD5 is the default value provided as an MD5 hash for a non-existent
+	// database.
+	zeroMD5 = "00000000000000000000000000000000"
+)
+
+type Downloader interface {
+	GetOutdatedEditions(ctx context.Context) ([]Metadata, error)
+	DownloadEdition(ctx context.Context, edition Metadata) error
+	MakeOutput() ([]byte, error)
```

Comment:
The name of this doesn't make it obvious to me what it does. I wonder if we could have a better name for it. I also wonder if we really need it?